x/ibc/20-transfer/module.go

```diff
@@ -26,84 +27,103 @@ var (
 	_ module.AppModuleBasic = AppModuleBasic{}
 )
 
+// AppModuleBasic is the 20-transfer appmodulebasic
 type AppModuleBasic struct{}
 
-// Name returns the IBC transfer ICS name
+// Name implements AppModuleBasic interface
 func (AppModuleBasic) Name() string {
-	return SubModuleName
+	return ModuleName
 }
 
+// RegisterCodec implements AppModuleBasic interface
 func (AppModuleBasic) RegisterCodec(cdc *codec.Codec) {
 	RegisterCodec(cdc)
 }
 
+// DefaultGenesis implements AppModuleBasic interface
 func (AppModuleBasic) DefaultGenesis() json.RawMessage {
 	return nil
 }
 
+// ValidateGenesis implements AppModuleBasic interface
 func (AppModuleBasic) ValidateGenesis(bz json.RawMessage) error {
 	return nil
 }
 
+// RegisterRESTRoutes implements AppModuleBasic interface
 func (AppModuleBasic) RegisterRESTRoutes(ctx context.CLIContext, rtr *mux.Router) {
 	rest.RegisterRoutes(ctx, rtr)
 }
 
+// GetTxCmd implements AppModuleBasic interface
```

Comment:
I would dearly love to know why all these implementations are valid no-op implementations. The comments as is just seem to check off a lint rule without providing much info. Why does `RegisterRESTRoutes` have to call `super` while `ValidateGenesis` can just return `nil`?

Reply:
`ValidateGenesis` handles validation for genesis data needed by the module. This module doesn't require anything in genesis, so `ValidateGenesis` doesn't need to do anything.

Reply:
`RegisterRESTRoutes` registers all the REST routes from this module onto the router for the `gaiacli rest-server`